src/movement/control_system.cpp

```diff
@@ -235,26 +235,51 @@ namespace robosub
                      * by adding in the new goal to the current goal to make it
                      * relative to the current state.
                      */
-                    this->goal_types[i] = robosub::control::STATE_ABSOLUTE;
-                    if(i < 3)
-                    {
-                        this->goals[i] = state_vector[i] + control_values[i];
-                    }
-                    else if (i == 3)
-                    {
-                        this->goals[i] = wraparound(state_vector[i] +
-                                control_values[i], -180.0, 180.0);
-                    }
-                    else if (i == 4)
+
+                    //if we just want to maintain our current state
+                    if(control_values[i] == 0)
                     {
-                        this->goals[i] = wraparound(state_vector[i] +
-                                control_values[i], -90.0, 90.0);
+                        //if the previous goal was absolute, we can just reuse
+                        //the last absolute goal
+                        if (this->goal_types[i] ==
+                            robosub::control::STATE_ABSOLUTE)
+                        {
+                            this->goals[i] = this->goals[i];
```

Comment:
I'm just being explicit, I suppose I could have an empty if statement here with a comment. I'll fix that.